scripts/test-runner.js

```diff
@@ -101,24 +98,21 @@ function runTest(name, code, args) {
     uniqueSuffix: "",
   };
   if (code.includes("// throws introspection error")) {
-    let onError = (realm, e) => {
-      if (IsIntrospectionError(realm, e)) throw new Success();
-    };
     try {
       let realmOptions = { serialize: true, compatibility, uniqueSuffix: "" };
       let realm = construct_realm(realmOptions);
       initializeGlobals(realm);
       let serializerOptions = { initializeMoreModules: speculate, delayUnsupportedRequires, internalDebug: true };
       let serializer = new Serializer(realm, serializerOptions);
       let sources = [{ filePath: name, fileContents: code }];
-      let serialized = serializer.init(sources, false, onError);
+      let serialized = serializer.init(sources, false);
```

Comment:
I think I had introduced the onError parameter to the serializer just for dealing with Introspection errors here. Can the concept be removed in the serializer code if there's no other user for it?